api/app/Modules/Library/Http/Controllers/DraftLyricsController.php

```diff
@@ -31,24 +30,58 @@ public function index(ListDraftLyricsRequest $request): JsonResponse
         return $this->respondWithItem($draftLyrics);
     }
 
-    public function lock(DraftLyrics $draftLyrics): Response
+    /**
+     * @throws AuthorizationException
+     */
+    public function lock(DraftLyrics $draftLyrics): JsonResponse
     {
-        $lock = Cache::get("draftLyricsForTrack:{$draftLyrics->track_id}");
-
-        if ($lock && $lock != Auth::id()) {
-            throw DraftUnavailableException::forEntity("Lyrics");
+        try {
+            $this->authorize('lock', $draftLyrics);
+            /** @phpstan-ignore-next-line  */
+        } catch ( DraftUnavailableException $e) {
+            return response()->json([
+                'error' => $e->getMessage()
+            ], 423);
         }
 
         Cache::put("draftLyricsForTrack:{$draftLyrics->track_id}", Auth::id(), 3600);
 
-        return response()->noContent();
+        return response()->json();
+    }
+
+    /**
+     * @throws AuthorizationException
+     */
+    public function unlock(DraftLyrics $draftLyrics): JsonResponse {
+        try {
+            $this->authorize('unlock', $draftLyrics);
+            /** @phpstan-ignore-next-line */
+        } catch (DraftUnavailableException $e) {
+            return response()->json([
+                'error' => $e->getMessage()
+            ], 423);
+        }
+
+        Cache::forget("draftLyricsForTrack:{$draftLyrics->track_id}");
+
+        return response()->json();
```

Comment:
So, the 423 response doesn't need to return the exception message either. It can just be a 423 response code with no content. The frontend only needs to check for that status code. 

So in that case, both of these would be a Response return type.

Reply:
Done, return type now Response :)